presto-hive/src/main/java/com/facebook/presto/hive/HiveCachingHdfsConfiguration.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.cache.CacheConfig;
+import com.facebook.presto.cache.CacheManager;
+import com.facebook.presto.cache.CacheStats;
+import com.facebook.presto.cache.CachingFileSystem;
+import com.facebook.presto.cache.ForCachingFileSystem;
+import com.facebook.presto.cache.LocalRangeCacheManager;
+import com.facebook.presto.cache.NoOpCacheManager;
+import com.facebook.presto.hadoop.FileSystemFactory;
+import com.facebook.presto.hive.HdfsEnvironment.HdfsContext;
+import com.facebook.presto.spi.PrestoException;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapred.JobConf;
+
+import javax.inject.Inject;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Map;
+import java.util.function.BiFunction;
+
+import static com.facebook.airlift.concurrent.Threads.daemonThreadsNamed;
+import static com.facebook.presto.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.Executors.newScheduledThreadPool;
+
+public class HiveCachingHdfsConfiguration
+        implements HdfsConfiguration
+{
+    private final HdfsConfiguration hiveHdfsConfiguration;
+    private final CacheManager cacheManager;
+    private final boolean cacheValidationEnabled;
+
+    @Inject
+    public HiveCachingHdfsConfiguration(
+            @ForCachingFileSystem HdfsConfiguration hdfsConfiguration,
+            CacheConfig cacheConfig,
+            CacheStats cacheStats)
+    {
+        this.hiveHdfsConfiguration = requireNonNull(hdfsConfiguration, "hiveHdfsConfiguration is null");
+
+        CacheConfig config = requireNonNull(cacheConfig, "cacheConfig is null");
+        this.cacheManager = config.getBaseDirectory() == null ?
```

Comment:
If we already created the `CacheManager` in module, maybe we could directly inject it, like the change you made in raptor?